lib/jekyll/converters/markdown.rb

```diff
@@ -1,54 +1,63 @@
 module Jekyll
   module Converters
     class Markdown < Converter
-      safe true
-
       highlighter_prefix "\n"
       highlighter_suffix "\n"
+      safe true
 
       def setup
         return if @setup
-        @parser =
-          case @config['markdown'].downcase
-            when 'redcarpet' then RedcarpetParser.new(@config)
-            when 'kramdown'  then KramdownParser.new(@config)
-            when 'rdiscount' then RDiscountParser.new(@config)
-          else
-            # So they can't try some tricky bullshit or go down the ancestor chain, I hope.
-            if allowed_custom_class?(@config['markdown'])
-              self.class.const_get(@config['markdown']).new(@config)
-            else
-              Jekyll.logger.error "Invalid Markdown Processor:", "#{@config['markdown']}"
-              Jekyll.logger.error "", "Valid options are [ #{valid_processors.join(" | ")} ]"
-              raise Errors::FatalException, "Invalid Markdown Processor: #{@config['markdown']}"
-            end
-          end
-        @setup = true
+        if (!@parser = get_processor)
+          Jekyll.logger.error "Invalid Markdown processor given:", @config["markdown"]
+          Jekyll.logger.info  "", "Custom processors are not loaded in safe mode" if @config["safe"]
+          Jekyll.logger.error "", "Available processors are: #{valid_processors.join(", ")}"
+          raise Errors::FatalException, "Bailing out; invalid Markdown processor."
+        end
+
+        @setup = \
+          true
+      end
+
+      def get_processor
+        case @config["markdown"].downcase
+          when "redcarpet" then return RedcarpetParser.new(@config)
+          when "kramdown"  then return KramdownParser.new(@config)
+          when "rdiscount" then return RDiscountParser.new(@config)
+        else
+          get_custom_processor
+        end
       end
 
+      # Public: Provides you with a list of processors, the ones we
+      # support internally and the ones that you have provided to us (if you
+      # are not in safe mode.)
+
       def valid_processors
-        %w[
-          rdiscount
-          kramdown
-          redcarpet
-        ] + third_party_processors
+        %W(rdiscount kramdown redcarpet) + \
+          third_party_processors
```

Comment:
They don't differ, it just happened automatically on my end and it's something I should probably remove in favor of the widely used `%w[]` for consistency.